src/game/engine/overlay.cpp

```diff
@@ -0,0 +1,196 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief
+ *
+ * @copyright Chronoshift is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "overlay.h"
+#include "cell.h"
+#include "iomap.h"
+#include "globals.h"
+
+#ifndef GAME_DLL
+TFixedIHeapClass<OverlayClass> g_Overlays;
+#endif
+
+HousesType OverlayClass::s_ToOwn = HOUSES_NONE;
+
+/**
+ *
+ *
+ */
+OverlayClass::OverlayClass(OverlayType type, cell_t cell, HousesType house) :
+    ObjectClass(RTTI_OVERLAY, g_Overlays.ID(this)),
+    m_Class(g_OverlayTypes.Ptr(type))
+{
+    if (cell != -1) {
+        s_ToOwn = house;
+        Unlimbo(Cell_To_Coord(cell));
+        s_ToOwn = HOUSES_NONE;
+    }
+}
+
+/**
+ *
+ *
+ */
+OverlayClass::OverlayClass(const OverlayClass &that) :
+    ObjectClass(that),
+    m_Class(that.m_Class)
+{
+}
+
+/**
+ *
+ *
+ */
+OverlayClass::OverlayClass(const NoInitClass &noinit) :
+    ObjectClass(noinit)
+{
+}
+
+/**
+ *
+ *
+ */
+OverlayClass::~OverlayClass()
+{
+    if (GameActive) {
+        OverlayClass::Limbo();
+    }
+    m_Class = nullptr;
+}
+
+/**
+ *
+ *
+ */
+void *OverlayClass::operator new(size_t size)
+{
+    OverlayClass *this_ptr = g_Overlays.Alloc();
+    if (this_ptr != nullptr) {
+        this_ptr->m_IsActive = true;
+    }
+    return this_ptr;
+}
+
+/**
+ *
+ *
+ */
+void OverlayClass::operator delete(void *ptr)
+{
+    OverlayClass *this_ptr = static_cast<OverlayClass *>(ptr);
+    if (this_ptr != nullptr) {
+        this_ptr->m_IsActive = false;
+    }
+    g_Overlays.Free(this_ptr);
+}
+
+/**
+ *
+ *
+ */
+void OverlayClass::Draw_It(int x, int y, WindowNumberType window) const
+{
+    //empty
+}
+
+/**
+ *
+ *
+ */
+BOOL OverlayClass::Mark(MarkType mark)
+{
+    if (!ObjectClass::Mark(mark) || mark != MARK_PUT) {
+        return false;
+    }
```

Comment:
Inconsistent white space, you put a new line in before/after some scopes but not all in the same function! Me nad CCHyper almost always have whitespace where scopes open and close in the same indentation level and don't where there is a chance in indentation.